test/functional/tests.py

```diff
@@ -807,6 +807,68 @@ def test_bitswap_unwant(self):
         result = self.api.bitswap_unwant(key='QmZTR5bcpQD7cFgTorqxZDYaew1Wqgfbd2ud9QqGPAkK2V')
         self.assertTrue(result is not None)
 
+@skipIfOffline()
+class IpfsApiPubSubTest(unittest.TestCase):
+
+    def setUp(self):
+        self.api = ipfsapi.Client()
+
+    def test_pubsub_pub_sub(self):
+        """
+        We should test both publishing and subscribing
+        at the same time for it's difficult
+        to determine if one works without checking the
+        other.
+        """
+        # the topic that will be published/subscribed to
+        topic = 'testing'
+        # the message that will be published
+        message = 'hello'
+
+        expected_data = 'aGVsbG8='	
+        expected_topicIDs = [topic]
+
+
+        # get the subscription stream
+        sub = self.api.pubsub_sub(topic)
+
+        # make sure something was actually returned from the subscription
+        assert sub is not None
+
+        # publish a message to topic
+        self.api.pubsub_pub(topic, message)
+
+        # get the 
+        sub_data = next(sub)
+
+        self.assertEqual(sub_data['data'], expected_data)
+        self.assertEqual(sub_data['topicIDs'], expected_topicIDs)
+
+    def test_pubsub_ls(self):
+        """
+        Testing the ls, assumes we are able
+        to at least subscribe to a topic
+        """
+        topic = 'lsing'
+        expected_return = { 'Strings': [topic] }
+
+        # subscribe to the topic testing
+        # in this test we don't care about the return value
+        a = self.api.pubsub_sub(topic)   
+
+        # grab the channels we're subscribed to
+        channels = self.api.pubsub_ls()
+        self.assertEqual(channels, expected_return)
+
+    def test_pubsub_peers(self):
+        """
+        Not sure how to test this since it fully depends
+        on who we're connected to. We may not even have
+        any peers
+        """
```

Comment:
Indeed :slightly_smiling_face: 
But please call the function anyways and check if its result appears to make sense (a list with some values probably), so that we can at least detect if the function is totally broken for some reason.

Reply:
Added a call to the function as well as a few assertions